src/api.ts

```
@@ -1,11 +1,18 @@
 import { MovieData } from './types/MovieData';
-import { ResponseError } from './types/ReponseError';
+import { ResponseError } from './types/ResponseError';
 
-const API_URL = 'https://www.omdbapi.com/?apikey=your-key';
+const API_KEY = process.env.REACT_APP_OMDB_API_KEY;
```

The API key is read from process.env. If the environment variable is missing this will make API_URL contain the string `undefined`, producing invalid requests at runtime. Please add a guard so you return/throw a clear error (or return a ResponseError) when the key is not set. Example approaches:
- If no key: return Promise.resolve({ Response: 'False', Error: 'Missing OMDb API key' });
- Or throw an Error so callers can treat it as network failure.
This will avoid confusing runtime behavior and make tests deterministic.